lib/spack/spack/spec.py

```diff
@@ -527,6 +527,13 @@ def __init__(self, spec_like, *dep_like, **kwargs):
             # XXX(deptype): default deptypes
             self._add_dependency(spec, ('build', 'link'))
 
+    def __getattr__(self, item):
+        """Delegate to self.package if the attribute is not in the spec"""
+        # This line is to avoid infinite recursion in case package is
+        # not present among self attributes
+        package = super(Spec, self).__getattribute__('package')
+        return getattr(package, item)
```

Comment:
@alalazo: if you're working on a fix I'll merge it asap, otherwise I'll work on it.